### ISSUE TITLE
[Project 2] Question about Fairness Policy

### ISSUE THREAD

**Issue by JinhwanSul** [swsnu/osspr2021#46]
Fairness Policy 에서 궁금한 점이 있어서 질문드립니다.
아래의 예시 상황처럼 writer 보다 Reader request 가 먼저 온 상황에서도 Fairness Policy 가 적용되는지 궁금합니다.

즉,
1st request : R[30, 50] -> granted
2nd request: W[10, 20] -> granted
3rd request : R[10, 30] -> sleep
4th request: W[0, 50] -> sleep

위 상황에서 W[10,20] 이 unlock 되면 3rd는 조건을 만족하고, 4th는 여전히 sleep입니다. 이 때, 3rd 는 4th 보다 request가 먼저 왔음에도 Fairness policy 때문에 sleep 을 해야하는 것인지, 먼저 왔으니까 granted 가 되는 것인지 궁금합니다.

**Comment by jaywonchung** [comment]
You're right. The third request remains sleeping, and the fourth request is granted write access.

As a side note, notice that if requests identical to the fourth one flood in, the third request will sleep forever. Thus this scenario proves the existence of "reader starvation". The given policy doesn't care about reader starvation.

If you want to know more, I recommend that you read [this article](https://en.m.wikipedia.org/wiki/Readers–writers_problem).
